nilearn/region_extraction/SimpleThresholdExtractor.py

```diff
@@ -0,0 +1,128 @@
+"""
+Voxel_ratio_extractor
+"""
+
+import nibabel
+import numpy as np
+
+from scipy.ndimage import label
+from scipy.stats import scoreatpercentile
+
+from nilearn import plotting
+from nilearn._utils import check_niimg
+from nilearn._utils.niimg_conversions import concat_niimgs
+from nilearn.image import iter_img, new_img_like
+
+
+def extract_seperate_regions(img, img_data, affine, min_size):
+    """ Extract the components which have largest
+        connections given a thresholded image of the
+        components
+
+        Parameters
+        ----------
+        img: numpy array of 3D Nifti like object
+            contains a thresholded image of the components
+
+        Returns
+        -------
+        the seperated or segmented regions extracted from
+        the input image which contains a set of connected
+        maps
+
+    """
+    all_regions_img = []
+    # n_components and n_labels before removing
+    # spurious regions
+    components, labels = label(img_data)
+    label_count = np.bincount(components.ravel())
+    label_count[0] = 0
+    for label_assign, count in enumerate(label_count):
+        if count < min_size:
+            img_data[components == label_assign] = 0
+
+    # Extract n_regions and n_labels survived after
+    # minimizing the unwanted or spurious regions
+    n_components, n_regions = label(img_data)
```

Comment:
You should not call `label` twice. This is a computationally expensive function.